eventsources/eventing.go

```diff
@@ -263,20 +268,90 @@ func NewEventSourceAdaptor(eventSource *v1alpha1.EventSource, eventBusConfig *ev
 
 // Start function
 func (e *EventSourceAdaptor) Start(ctx context.Context) error {
-	logger := logging.FromContext(ctx).Desugar()
-	logger.Info("Starting event source server...")
+	log := logging.FromContext(ctx)
+
+	recreateTypes := make(map[apicommon.EventSourceType]bool)
+	for _, esType := range apicommon.RecreateStrategyEventSources {
+		recreateTypes[esType] = true
+	}
+	isRecreatType := false
 	servers := GetEventingServers(e.eventSource, e.metrics)
+	for k := range servers {
+		if _, ok := recreateTypes[k]; ok {
+			isRecreatType = true
+		}
+		// This is based on the presumption that all the events in one
+		// EventSource object use the same type of deployment strategy
+		break
+	}
+	if !isRecreatType || e.eventSource.Spec.GetReplicas() == 1 {
```

Comment:
Minor - is workflows we always use leader election.  What happens when you scale from 1 to 2 and the first replica is not already using it? Don’t you get two leaders?

Reply:
I would also recommend using leader election by default which will avoid the code complexity to deal separately. Workflow enables election leader by default

Reply:
Ah - so users must all install the extra RBAC - a process that does not scale. Good point.